controllers/license/konglicense_controller.go

```diff
@@ -317,23 +427,19 @@ func (r *KongV1Alpha1KongLicenseReconciler) ensureControllerStatusProgrammedCond
 
 	fullControllerName := r.fullControllerName()
 	// Find the managed controller status item and append new item when absent.
-	controllerStatus, controllerIndex, found := lo.FindIndexOf(license.Status.KongLicenseControllerStatuses, func(controllerStatus kongv1alpha1.KongLicenseControllerStatus) bool {
+	_, controllerIndex, found := lo.FindIndexOf(license.Status.KongLicenseControllerStatuses, func(controllerStatus kongv1alpha1.KongLicenseControllerStatus) bool {
 		return controllerStatus.ControllerName == fullControllerName
 	})
 	if !found {
 		wantedControllerStatus := kongv1alpha1.KongLicenseControllerStatus{
 			ControllerName: fullControllerName,
-			Conditions: []metav1.Condition{
-				{
-					Type:               ConditionTypeProgrammed,
-					LastTransitionTime: metav1.Now(),
-					Status:             programmedStatus,
-					Reason:             reason,
-				},
-			},
 		}
 		license.Status.KongLicenseControllerStatuses = append(license.Status.KongLicenseControllerStatuses, wantedControllerStatus)
-		return r.Client.Status().Update(ctx, license)
+		controllerIndex = 0
+	}
+	ctrlManagedConditions := &license.Status.KongLicenseControllerStatuses[controllerIndex].Conditions
+	if r.LicenseValidator != nil {
+		setLicenseValidityCondition(ctrlManagedConditions, r.LicenseValidator(license.RawLicenseString))
 	}
```

Comment:
Please consider renaming `ensureControllerStatusProgrammedCondition` method - now it sets both `Programmed` and `LicenseValid` conditions so the old naming doesn't match.